### PR TITLE
EP: guardar history post save

### PR DESCRIPTION
**Pull request by plammel**
### Requerimiento

Refactor de historial de ficha: Se cambia el schema a el mismo de la Ficha y se guarda el registro en middleware post save
### Funcionalidad desarrollada 

1. 
2. 
3. 

### UserStories llegó a completarse

- [ ] Si
- [ ] No

### Requiere actualizaciones en la base de datos

- [x] Si: *NOTA:* Dado los cambios de schema en el PR, el script debe correrse en el shell de mongo.
[script.update.fichaHistorial (1).txt](https://github.com/andes/api/files/6939810/script.update.fichaHistorial.1.txt)
- [ ] No

### CODE REVIEW

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4711](http://condorpiedra.andes.gob.ar:85/4711/mochawesome.html)
CYPRESS RUN: [3710](https://dashboard.cypress.io/projects/xr7gft/runs/3710/specs)
TEST START: 2021-07-26T18:39:45.686Z
TOTAL: 375
SUCCESS: 362
FAIL: 12
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4727](http://condorpiedra.andes.gob.ar:85/4727/mochawesome.html)
CYPRESS RUN: [3724](https://dashboard.cypress.io/projects/xr7gft/runs/3724/specs)
TEST START: 2021-07-27T14:13:09.038Z
TOTAL: 392
SUCCESS: 389
FAIL: 2
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4733](http://condorpiedra.andes.gob.ar:85/4733/mochawesome.html)
CYPRESS RUN: [3730](https://dashboard.cypress.io/projects/xr7gft/runs/3730/specs)
TEST START: 2021-07-27T18:12:58.300Z
TOTAL: 382
SUCCESS: 378
FAIL: 3
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4738](http://condorpiedra.andes.gob.ar:85/4738/mochawesome.html)
CYPRESS RUN: [3735](https://dashboard.cypress.io/projects/xr7gft/runs/3735/specs)
TEST START: 2021-07-28T12:15:21.064Z
TOTAL: 400
SUCCESS: 397
FAIL: 2
SKIPPED: 1

**Review by negro89 — APPROVED** [review]

**Comment by liquid36** [comment]
USUARIO: lmlagos  
BUILD NUMBER: [4757](http://condorpiedra.andes.gob.ar:85/4757/mochawesome.html)
CYPRESS RUN: [3753](https://dashboard.cypress.io/projects/xr7gft/runs/3753/specs)
TEST START: 2021-07-29T15:54:20.221Z
TOTAL: 400
SUCCESS: 398
FAIL: 1
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4781](http://condorpiedra.andes.gob.ar:85/4781/mochawesome.html)
CYPRESS RUN: [3777](https://dashboard.cypress.io/projects/xr7gft/runs/3777/specs)
TEST START: 2021-08-03T15:03:27.401Z
TOTAL: 400
SUCCESS: 398
FAIL: 1
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4786](http://condorpiedra.andes.gob.ar:85/4786/mochawesome.html)
CYPRESS RUN: [3782](https://dashboard.cypress.io/projects/xr7gft/runs/3782/specs)
TEST START: 2021-08-03T16:16:17.221Z
TOTAL: 400
SUCCESS: 398
FAIL: 1
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4788](http://condorpiedra.andes.gob.ar:85/4788/mochawesome.html)
CYPRESS RUN: [3784](https://dashboard.cypress.io/projects/xr7gft/runs/3784/specs)
TEST START: 2021-08-03T16:57:30.792Z
TOTAL: 334
SUCCESS: 317
FAIL: 16
SKIPPED: 1

**Comment by liquid36** [comment]
USUARIO: plammel  
BUILD NUMBER: [4792](http://condorpiedra.andes.gob.ar:85/4792/mochawesome.html)
CYPRESS RUN: [3787](https://dashboard.cypress.io/projects/xr7gft/runs/3787/specs)
TEST START: 2021-08-03T18:07:01.223Z
TOTAL: 400
SUCCESS: 399
FAIL: 0
SKIPPED: 1

**Review by martinebucarey — APPROVED** [review]

File: modules/forms/forms-epidemiologia/forms-history.schema.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
-import { AuditPlugin } from '@andes/mongoose-plugin-audit';
 import * as mongoose from 'mongoose';
 import { FormsEpidemiologiaCloneSchema } from './forms-epidemiologia-schema';
 
 
-export const FormsHistorySchema = new mongoose.Schema({
-    ficha: FormsEpidemiologiaCloneSchema
+const FormsHistorySchema = new mongoose.Schema({
```
Comment (liquid36) [inline modules/forms/forms-epidemiologia/forms-history.schema.ts:5]:
```suggestion
const FormsHistorySchema = FormsEpidemiologiaCloneSchema.clone().add({
    id: ...
})
```

File: modules/forms/forms-epidemiologia/forms-history.schema.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+    ...FormsEpidemiologiaCloneSchema.obj,
+    createdAt: Date,
+    createdBy: { nombreCompleto: String, username: String },
+    updatedAt: Date,
+    updatedBy: { nombreCompleto: String, username: String },
```
Comment (liquid36) [inline modules/forms/forms-epidemiologia/forms-history.schema.ts:10]: Estos campos ya deberían venir con la ficha.

File: modules/forms/forms-epidemiologia/forms-epidemiologia-schema.ts [hunk 3]
```diff
@@ -68,5 +70,21 @@ FormsEpidemiologiaSchema.pre('save', function (next) {
     next();
 });
 
+FormsEpidemiologiaSchema.post('save', (ficha: any) => {
+    const { FormsHistory } = require ('./forms-history.schema');
+    let history = new FormsHistory();
```
Comment (liquid36) [inline modules/forms/forms-epidemiologia/forms-epidemiologia-schema.ts:75]:
```suggestion
    const history = new FormsHistory(
        ficha.toJSON()
    );
```

File: modules/forms/forms-epidemiologia/forms-epidemiologia-schema.ts [hunk 4]
```diff
@@ -68,5 +70,21 @@ FormsEpidemiologiaSchema.pre('save', function (next) {
...
+
+    history.id = ficha.id;
+    history.paciente = ficha.paciente;
+    history.secciones = ficha.secciones;
+    history.type = ficha.type;
+    history.createdAt = ficha.createdAt;
+    history.updatedAt = ficha.updatedAt;
+    history.createdBy = ficha.createdBy;
+    history.updatedBy = ficha.updatedBy;
```
Comment (liquid36) [inline modules/forms/forms-epidemiologia/forms-epidemiologia-schema.ts:84]: y todo esto no iría...

**Pull request merged**